### PR TITLE
Add xml support for get_static_var

### LINKED ISSUE #515: Support `xml` file with multiple variables
Modify `get_static_variable` in `e3sm_diags/driver/utils/dataset.py` to support a `xml` file with multiple variables. This is necessary for https://github.com/E3SM-Project/zppy/pull/126

### PR DESCRIPTION
**Pull request by forsyth2**
Add multi-variable support for `xml`. Resolves #515.

### CODE REVIEW

File: e3sm_diags/driver/utils/dataset.py [hunk 1]
```diff
@@ -179,6 +179,14 @@ def get_static_variable(self, static_var, primary_var):
...
             data_path = self.parameters.test_data_path
         file_path = self._get_timeseries_file_path(primary_var, data_path)
+        # Works:
+        #file_path=/lcrc/group/e3sm/public_html/e3sm_diags_test_data/unit_test_complete_run/test/time-series/streamflow_ts_postprocessed/RIVER_DISCHARGE_OVER_LAND_LIQ_198001_201412.nc
+        #static_var=areatotal2
+        # Fails:
+        #file_path=rof_links/RIVER_DISCHARGE_OVER_LAND_LIQ_198501_201412.xml
+        #static_var=areatotal2
+        print("file_path={}".format(file_path))
+        print("static_var={}".format(static_var))
```
Comment (forsyth2) [inline e3sm_diags/driver/utils/dataset.py:189]: @chengzhuzhang It seems like the issue is `cdms2.open` can't read a `xml` file. So do we need to change what E3SM Diagnostics uses or do we need to make the fix in CDMS?

Comment (chengzhuzhang) [reply e3sm_diags/driver/utils/dataset.py:189]: It is weird, cdms2.open does open xml files. You could try on other single variable .xml files. I guess the problem is if cdms2.open reads a xml file with multiple variables.

Comment (chengzhuzhang) [reply e3sm_diags/driver/utils/dataset.py:189]: actually it work as follows:
```
file = 'RIVER_DISCHARGE_OVER_LAND_LIQ_198501_201412.xml'
import cdms2
fin = cdms2.open(file)
var = fin('RIVER_DISCHARGE_OVER_LAND_LIQ')
var = fin('areatotal2')
```

**Review by forsyth2 — COMMENTED** [review]
@chengzhuzhang This is ready for review. Is it ok to `cdms2.open` without a corresponding `close`? I thought the `with` pattern was specifically to close resources after use.

**Comment by forsyth2** [comment]
@tomvothecoder I had to use `--no-verify` on the commit. Without it, I get `An unexpected error has occurred: CalledProcessError: command: ('/home/ac.forsyth2/miniconda3/envs/e3sm_diags_env_dev_20210827/bin/python', '-mvirtualenv', '/gpfs/fs1/home/ac.forsyth2/.cache/pre-commit/repoe318f1p1/py_env-python3.9')` followed by stack traces ending in:

```
BlockingIOError: [Errno 11] Resource temporarily unavailable: '/gpfs/fs1/home/ac.forsyth2/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/setuptools-57.4.0-py3-none-any/setuptools/gui-32.exe' -> '/gpfs/fs1/home/ac.forsyth2/.cache/pre-commit/repoe318f1p1/py_env-python3.9/lib/python3.9/site-packages/setuptools/./gui-32.exe'
```

and 

```
BlockingIOError: [Errno 11] Resource temporarily unavailable: '/gpfs/fs1/home/ac.forsyth2/.local/share/virtualenv/wheel/3.9/image/1/CopyPipInstall/pip-21.1.2-py3-none-any/pip/_vendor/distlib/w32.exe' -> '/gpfs/fs1/home/ac.forsyth2/.cache/pre-commit/repoe318f1p1/py_env-python3.9/lib/python3.9/site-packages/pip/_vendor/distlib/w32.exe'
```

**Review by chengzhuzhang — COMMENTED** [review]
Yeah, it makes sense to use this open statement for xml files, you would need to close fin in this case.

**Comment by forsyth2** [comment]
> you would need to close fin in this case.

So would the following be better?
```
fin = cdms2.open(file_path)
result = fin(static_var)
cdms2.close(fin)
return result
```

**Comment by chengzhuzhang** [comment]
yep. `cdms2.close(fin)` or `fin.close()`

**Comment by chengzhuzhang** [comment]
I edited the PR title to be more accurate...

**Comment by forsyth2** [comment]
> fin.close()
It has to be that one. `cdms2.close(fin)` doesn't work.

>  I had to use --no-verify on the commit. 
The pre-commits-hooks CI check passes, so I view that as acceptable.

**Pull request merged**